observability/exporting.go

```diff
@@ -0,0 +1,141 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	gcplogging "cloud.google.com/go/logging"
+	grpclogrecordpb "google.golang.org/grpc/observability/internal/logging"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// loggingExporter is the interface of logging exporter for gRPC
+// Observability. Ideally, we should use what OTEL provides, but their Golang
+// implementation is in "frozen" state. So, this plugin provides a minimum
+// interface to satisfy testing purposes.
+type loggingExporter interface {
+	// EmitGrpcLogRecord writes a gRPC LogRecord to cache without blocking.
+	EmitGrpcLogRecord(*grpclogrecordpb.GrpcLogRecord)
+	// Close flushes all pending data and closes the exporter.
+	Close() error
+}
+
+// globalLoggingExporter is the global logging exporter, may be nil.
+var globalLoggingExporter loggingExporter
```

Comment:
Why isn't this a `*cloudLoggingExporter`?

Reply:
> In future, I hope we can allow users to bring their own exporters. But we don't have a design yet.

What would this buy them that implementing their own `binarylog.Sink` doesn't already?

Reply:
In the sink, we convert `binarylog.GrpcLogEntry` to `logging.GrpcLogRecord`. Then in the exporter, we convert `logging.GrpcLogRecord` to `google.logging.LogEntry`.

By using our sink, users gain the access to directly work with `logging.GrpcLogRecord`, if it will be our future canonical data model.